app/controllers/auth/registrations_controller.rb

```diff
@@ -28,7 +28,8 @@ def after_inactive_sign_up_path_for(_resource)
   end
 
   def check_enabled_registrations
-    redirect_to root_path if Rails.configuration.x.single_user_mode || !Setting.open_registrations
+    user_count = Rails.cache.fetch('user_count') { User.count }
+    redirect_to root_path if Rails.configuration.x.single_user_mode || !Setting.open_registrations || user_count >= (Setting.max_users || Float::INFINITY)
```

Comment:
would this not be cleaner to either use `InstancePresenter.new.open_registrations` or move into a model (settings?) so its not duplicated?